shopinvader_company_backend/models/product_product.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Camptocamp (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _name = "product.product"
+    _inherit = ["product.product", "shopinvader.company.backend.mixin"]
+
+    # Warning: These fields are only intended to be used from the UI.
+    # Result will not be what you expect if used with with_company,
+    # due to their cache not being invalidated.
+
+    company_shopinvader_bind_ids = fields.One2many(
+        "shopinvader.variant",
+        "record_id",
+        domain=lambda self: self._get_shopinvader_company_binding_domain(),
+        readonly=True,
+    )
+    company_shopinvader_published = fields.Boolean(
+        string="Published on Shopinvader",
+        compute="_compute_company_shopinvader_published",
+        inverse="_inverse_company_shopinvader_published",
+        search="_search_company_shopinvader_published",
+    )
+
+    def _shopinvader_bind(self, backend):
+        BindingWizard = self.env["shopinvader.variant.binding.wizard"]
+        wizard = BindingWizard.create(
+            {
+                "backend_id": backend.id,
+                "product_ids": [(6, 0, self.ids)],
+                "run_immediately": True,
+            }
+        )
+        wizard.bind_products()
```

Comment:
Note you can also can directly the method "bind_selected_products" of backend, as the wizard will just call this method

```suggestion
        backend.bind_selected_products(self, run_immediately=True)
```